spec/requests/voucher_adjustments_spec.rb

```diff
@@ -4,55 +4,120 @@
 
 describe VoucherAdjustmentsController, type: :request do
   let(:user) { order.user }
+  let(:address) { create(:address) }
   let(:distributor) { create(:distributor_enterprise, with_payment_and_shipping: true) }
-  let(:order) { create( :order_with_line_items, line_items_count: 1, distributor: distributor) }
+  let(:order_cycle) { create(:order_cycle, distributors: [distributor]) }
+  let(:exchange) { order_cycle.exchanges.outgoing.first }
+  let(:order) do
+    create(
+      :order_with_line_items,
+      line_items_count: 1,
+      distributor: distributor,
+      order_cycle: order_cycle,
+      bill_address: address,
+      ship_address: address
+    )
+  end
+  let(:shipping_method) { distributor.shipping_methods.first }
   let(:voucher) { create(:voucher, code: 'some_code', enterprise: distributor) }
-  let!(:adjustment) { voucher.create_adjustment(voucher.code, order) }
 
   before do
     # Make sure the order is created by the order user, the factory doesn't set ip properly
     order.created_by = user
     order.save!
 
+    exchange.variants << order.line_items.first.variant
```

Comment:
I am lacking context around exchange I think. If I understand correctly, this add the first line item in the order to the exchange (which is an Enterprise from the order cycle point of view ) , is that correct ? If so why is it needed ? and shouldn't the factory `:order_with_line_items` set that up for us ?

Reply:
Yeah it looks like this line can be dropped :+1: